examples/13-agent-name-service/agent1.py

```diff
@@ -0,0 +1,44 @@
+from cosmpy.aerial.wallet import LocalWallet
+
+from uagents.network import get_ledger, get_name_service_contract
+from uagents.setup import fund_agent_if_low
+from uagents import Agent, Context, Model
+
+
+# NOTE: Run agent1.py before running agent2.py
+
+
+class Message(Model):
+    message: str
+
+
+bob = Agent(
+    name="bob-0",
+    seed="agent bob-0 secret phrase",
+    port=8001,
+    endpoint=["http://localhost:8001/submit"],
+)
+
+ledger = get_ledger()
+my_wallet = LocalWallet.from_unsafe_seed("registration test wallet")
+name_service_contract = get_name_service_contract()
+DOMAIN = "agent"
+
+for wallet in [my_wallet, bob.wallet]:
+    fund_agent_if_low(wallet.address())
+
+
+@bob.on_event("startup")
+async def register_agent_name(ctx: Context):
+    await name_service_contract.register(
```

Comment:
I see that the name service registration is not handled implicitly by the library unlike almanac registration.
Not sure if it is better to have it as part of the agent handlers or to put it outside similar to `agent.run()`

Reply:
Right, we originally had it inside the agent similar to the Almanac registration, but decided to move it out because it seemed to make more sense when being sent from an external wallet.